ui/src/auth/AuthProviderDefault.js

```diff
@@ -64,63 +68,188 @@ export const AuthProvider = ({ children }) => {
     });
   };
 
+  const setIsSessionExpired = (isSessionExpired) => {
+    dispatch({
+      type: "SESSION_EXPIRED",
+      isSessionExpired,
+    });
+  };
+
+  const sendRequestCommon = async (
+    json,
+    location = window.location.href,
+    method = "post"
+  ) => {
+    let body = null;
+
+    if (json) {
+      body = JSON.stringify(json);
+    }
+
+    if (state.isSessionExpired) {
+      return null;
+    }
+
+    const xsrf = getCookie("_xsrf");
+    return fetch(location, {
+      method: method,
+      headers: {
+        "Content-type": "application/json",
+        "X-Xsrftoken": xsrf,
+        "X-Requested-With": "XMLHttpRequest",
+        Accept: "application/json",
+      },
+      body: body,
+    })
+      .then((response) => {
+        if (
+          response.status === 403 &&
+          response.type === "redirect" &&
+          response.reason === "unauthenticated"
+        ) {
+          fetch("/auth?redirect_url=" + window.location.href, {
+            headers: {
+              "X-Requested-With": "XMLHttpRequest",
+              Accept: "application/json",
+            },
+          }).then((resp) => {
+            if (resp.type === "redirect") {
+              window.location.href = resp.redirect_url;
+            }
+          });
+        } else if (response.status === 401) {
+          setIsSessionExpired(true);
+        }
+
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw new Error(response);
+        }
+      })
+      .then((data) => {
+        return data;
+      })
+      .catch((error) => {
+        console.error(`Exception Raised: ${error}`);
+        return null;
+      });
+  };
+
+  const sendRequestV2 = async (requestV2) => {
```

Comment:
This is only used for policy requests / selfService? Should it be renamed sendPolicyRequest?